src/actions/emojis.js

```diff
@@ -43,14 +47,14 @@ export function getCustomEmoji(emojiId) {
     );
 }
 
-export function getCustomEmojiByName(name) {
+export function getCustomEmojiByName(name: string): ActionFunc {
     return async (dispatch, getState) => {
         const serverVersion = Client4.getServerVersion();
         if (!isMinimumServerVersion(serverVersion, 4, 7)) {
             return {data: {}};
         }
 
-        dispatch({type: EmojiTypes.GET_CUSTOM_EMOJI_REQUEST}, getState);
+        dispatch({type: EmojiTypes.GET_CUSTOM_EMOJI_REQUEST, data: null}, getState);
```

Comment:
I suppose the `data: null` here is required to satisfy Flow? This seems potentially cumbersome if required everywhere. Is it not possible to mark the parameter as optional and allow `undefined`?

Reply:
Yes, we can support optional data. But I'm not sure about it, I like the idea of explicitly data equal null, declared. But Flow supports optional fields.

Reply:
Anyway if we declare the data as optional here, we have to check in every single reducer the data existence before we use it. In the future, we can define for each action a type and make it very robust, and allow to only pass data when needed.

Reply:
@jespino Wouldn't we need to do a `null` check anyhow?